### ISSUE TITLE
libgd: url parser is broken for this package

### ISSUE THREAD

**Issue by davydden** [LLNL/spack#3176]
the actual download is: `https://github.com/libgd/libgd/releases/download/gd-2.2.4/libgd-2.2.4.tar.gz` however
```
==> Fetching https://github.com/libgd/libgd/releases/download/2.2.4/lib2.2.4.tar.gz

curl: (22) The requested URL returned error: 404 Not Found
==> Fetching from https://github.com/libgd/libgd/releases/download/2.2.4/lib2.2.4.tar.gz failed.
==> Error: FetchError: All fetchers failed for libgd-2.2.4-me7ixfjderv2zdr5r3nbfef47elpzi7l
```

**Comment by davydden** [comment]
@adamjstewart can you please check if https://github.com/LLNL/spack/pull/2972 fixes this issue? If so, please mark the OP as such.

**Comment by adamjstewart** [comment]
Before:
```
$ spack url parse https://github.com/libgd/libgd/releases/download/gd-2.2.4/libgd-2.2.4.tar.gz
==> Parsing URL: https://github.com/libgd/libgd/releases/download/gd-2.2.4/libgd-2.2.4.tar.gz

==> Matched version regex  7: r'github.com/[^/]+/[^/]+/releases/download/v?([^/]+)/.*$'
==> Matched  name   regex  8: r'([^/]+)gd-2.2.4'

==> Detected:
    https://github.com/libgd/libgd/releases/download/gd-2.2.4/libgd-2.2.4.tar.gz
                                                     ~~~~~~~~ ---
    name:    lib
    version: gd-2.2.4

==> Substituting version 9.9.9b:
    https://github.com/libgd/libgd/releases/download/9.9.9b/lib9.9.9b.tar.gz
                                                     ~~~~~~ ---
```
After #2972:
```
$ spack url parse https://github.com/libgd/libgd/releases/download/gd-2.2.4/libgd-2.2.4.tar.gz
==> Parsing URL: https://github.com/libgd/libgd/releases/download/gd-2.2.4/libgd-2.2.4.tar.gz

==> Matched version regex  0: r'^[a-zA-Z+._-]+[._-]v?(\\d[\\d._-]*)$'
==> Matched  name   regex  0: r'github\\.com/[^/]+/([^/]+)'

==> Detected:
    https://github.com/libgd/libgd/releases/download/gd-2.2.4/libgd-2.2.4.tar.gz
                             -----                                  ~~~~~
    name:    libgd
    version: 2.2.4

==> Substituting version 9.9.9b:
    https://github.com/libgd/libgd/releases/download/gd-9.9.9b/libgd-9.9.9b.tar.gz
                             -----                                   ~~~~~~
```
Looks like it will be fixed. I'll update #2972 to mention this fact.

**Comment by samfux84** [comment]
```
==> Fetching https://github.com/libgd/libgd/releases/download/2.2.4/lib2.2.4.tar.gz

curl: (35) error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol
==> Fetching from https://github.com/libgd/libgd/releases/download/2.2.4/lib2.2.4.tar.gz failed.
==> Error: FetchError: All fetchers failed for libgd-2.2.4-yfhkurdu5zwttbhcndqtrngefmzpx75w
```

The curl that is used to fetch the tarball has been built with spack using the system openssl from CentOs 7.2:

```
Concretized
--------------------------------
curl@7.52.1%gcc@4.8.5 cflags="-O3 -ftree-vectorize -mavx -fPIC -march=corei7-avx" cppflags="-O3 -ftree-vectorize -mavx -fPIC -march=corei7-avx" cxxflags="-O3 -ftree-vectorize -mavx -fPIC -march=corei7-avx"  arch=linux-centos7-x86_64 
    ^openssl@1.0.1e%gcc@4.8.5 cflags="-O3 -ftree-vectorize -mavx -fPIC -march=corei7-avx" cppflags="-O3 -ftree-vectorize -mavx -fPIC -march=corei7-avx" cxxflags="-O3 -ftree-vectorize -mavx -fPIC -march=corei7-avx"  arch=linux-centos7-x86_64 
    ^zlib@1.2.11%gcc@4.8.5 cflags="-O3 -ftree-vectorize -mavx -fPIC -march=corei7-avx" cppflags="-O3 -ftree-vectorize -mavx -fPIC -march=corei7-avx" cxxflags="-O3 -ftree-vectorize -mavx -fPIC -march=corei7-avx" +pic+shared arch=linux-centos7-x86_64 
```

**Comment by adamjstewart** [comment]
Yep, this will be fixed by #2972. I'm still putting the finishing touches in.
